### PR TITLE
update: Really basic SubmissionWorkflow banner

### LINKED ISSUE #1616: Render Submissions layout block for Collections with submission enabled
When rendering a Collection layout where the Collection has submissions enabled, grab the block content and render a new block into the layout in the appropriate place: at the top of the blocks unless there is a `CollectionHeader` in the first position, then under that.

TBD a little bit: do we reuse the banner block or create new block type for this specifically.

### PR DESCRIPTION
**Pull request by idreyn**
Kind of resolves #1616 

This is mostly code that glues together stuff we already have.

- Causes the workflow editor on the Submissions tab to actually create a workflow when "Save and Continue" is clicked.
- When a workflow is present and enabled a very basic looking submissions banner will appear in its layout. It will have a Submit button that can be clicked to create a new Pub

Lots of work left to do here, this is just enough code to help us test as we build other parts of Submissions.
 
image

_Test plan:_
Make sure the above things work :)

### CODE REVIEW

**Review by qweliant — APPROVED** [review]
This works a bit to well, i can just write pubs all willy nilly, LGTM

File: client/containers/DashboardSubmissions/NewSubmissionWorkflowEditor.tsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { Button } from '@blueprintjs/core';
 
+import { SubmissionWorkflow } from 'types';
 import { useLocalStorage } from 'client/utils/useLocalStorage';
 import { getEmptyDoc } from 'client/components/Editor';
 import { usePageContext } from 'utils/hooks';
 
+import { apiFetch } from 'client/utils/apiFetch';
```
Comment (3mcd) [inline client/containers/DashboardSubmissions/NewSubmissionWorkflowEditor.tsx:9]: Should this go up one import block?

File: server/routes/collection.tsx [hunk 2]
```diff
@@ -29,13 +32,42 @@ const findCollectionByPartialId = (maybePartialId) => {
...
 
+const getLayoutWithSubmissionWorkflowBlock = async (collection: types.Collection) => {
+	const { layout, id: collectionId } = collection;
+	if (layout) {
+		const workflow = await getEnabledSubmissionWorkflowForCollection(collectionId);
+		if (workflow?.enabled) {
+			const [firstBlock] = layout.blocks;
+			const { title, introText, id: submissionWorkflowId } = workflow;
+			const firstBlockIsHeader = firstBlock?.type === 'collection-header';
+			const bannerBlockIndex = firstBlockIsHeader ? 1 : 0;
```
Comment (3mcd) [inline server/routes/collection.tsx:50]: Must... resist... unary `+` operator...

**Review by 3mcd — APPROVED** [review]
LGTM just one nit about import ordering

**Pull request merged**